api/main.py

```diff
@@ -1,20 +1,57 @@
 from fastapi import FastAPI, Request, status
 from fastapi.encoders import jsonable_encoder
-from fastapi.exceptions import RequestValidationError
+from fastapi.exceptions import RequestValidationError, HTTPException
 from fastapi.responses import JSONResponse
 # from controller import Controller
+from starlette.responses import PlainTextResponse
+
+from object_store import ObjectStore
+import boto3
 
 import pickle
-import json
+import json, os
+import logging
 
 import numpy
 
+COMPLETED_BUCKET = os.getenv("COMPLETED_BUCKET")
+PENDING_BUCKET = os.getenv("PENDING_BUCKET")
+
+# Initialize ObjectStore
+try:
+    ob = ObjectStore()
+except Exception as ex:
+    logging.error("Error is -", ex)
+
+
 app = FastAPI()
 
 @app.get("/")
 async def homepage(request: Request):
     return("Welcome to Async Job Framework")
 
+
+@app.get("/getResult/{job_id}")
+async def getResult(job_id: str):
+    """Get the job id and try fetching the result."""
+    # TODO add error response for user, add logic to check in pending bucket if not found in completed.
+    job_id_extension = str(job_id).strip()+".json"
+    try:
+        result = ob.get_object(job_id_extension,COMPLETED_BUCKET)
+        #check if result is available or not in completed bucket and if not available check for the job in pending bucket
+        if result is None:
+            result = ob.get_object(job_id_extension,PENDING_BUCKET)
+            if result is None:
+                return HTTPException(status_code=404, detail="Item not found")
+            return HTTPException(status_code=201, detail="Result not available,job still pending")
```

Comment:
Instead of returning an exception, return the latest job status from the incomplete bucket.

Reply:
I have changed it to JSON responses, I guess once we implement the backend it would be a good idea to check status or fetch result if available. Maybe add that in a different PR?

Reply:
That's fine. Please open an issue for the change.

Reply:
https://github.com/CS6620-S21/Asynchronous-quantum-job-submission-framework/issues/20